src/cobalt.rs

```diff
@@ -20,6 +24,45 @@ use config::SassOutputStyle;
 use files::FilesBuilder;
 use frontmatter;
 
+fn deep_insert(data_map: &mut HashMap<String, Value>,
+               file_path: &Path,
+               target_key: String,
+               data: Value)
+               -> Result<()> {
+    // now find the nested map it is supposed to be in
+    let target_map = if let Some(path) = file_path.parent() {
+        let mut map = data_map;
+        for part in path.iter() {
+            if part.len() == 0 {
```

Comment:
This was a problem in an earlier version, with split, with top-level items. But you are right, this isn't a problem since the introduction of `path.iter()` anymore. removed.